Error message while fresh install without secret.py

We get an error message that is hard to figure out, when you are new to the project.
Fresh install of project. 
1. `npm run setup` 
2. `npm run pip`
4. `npm run start`

without creating a config yet (so there is `npm run config` missing as 3rd step).

Probably there is just a catch missing @karlcow @miketaylr ?


```
Statuses Initialization…
Fetching milestones from Github…
It failed with 404 Client Error: Not Found for url: https://api.github.com/repos/milestones!
We will read from data/milestones.json.

Traceback (most recent call last):
  File "run.py", line 35, in <module>
    from webcompat import app
  File "/Users/madeleine/projects/webcompat.com/webcompat/__init__.py", line 20, in <module>
    app.config.from_object('config')
  File "/Users/madeleine/projects/webcompat.com/env/lib/python2.7/site-packages/flask/config.py", line 168, in from_object
    obj = import_string(obj)
  File "/Users/madeleine/projects/webcompat.com/env/lib/python2.7/site-packages/werkzeug/utils.py", line 418, in import_string
    __import__(import_name)
  File "/Users/madeleine/projects/webcompat.com/config/__init__.py", line 132, in <module>
    if not initialize_status():
  File "/Users/madeleine/projects/webcompat.com/config/__init__.py", line 57, in initialize_status
    app.config['STATUSES'] = convert_milestones(milestones_content)
  File "/Users/madeleine/projects/webcompat.com/config/__init__.py", line 69, in convert_milestones
    STATUSES[milestone['title']]['id'] = milestone['number']
TypeError: string indices must be integers
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webcompat@ start: `source env/bin/activate || . env/bin/activate && python run.py`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the webcompat@ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/madeleine/.npm/_logs/2017-11-15T14_25_28_052Z-debug.log
```

Let's try to test with config done.

``` 
npm run start
```

This has **never** worked for me.

```
→ npm run start

> webcompat@ prestart /Users/karl/code/webcompat.com
> npm run build


> webcompat@ build /Users/karl/code/webcompat.com
> grunt

Running "checkDependencies:default" (checkDependencies) task

Running "jst:compile" (jst) task
File webcompat/static/js/templates.js created.

Running "concat:dist" (concat) task

Running "concat:diagnose" (concat) task

Running "concat:issues" (concat) task

Running "concat:issueList" (concat) task

Running "concat:userActivity" (concat) task

Running "uglify:dist" (uglify) task
>> 1 file created 548.27 kB → 389.51 kB

Running "uglify:issues" (uglify) task
>> 1 file created 45.23 kB → 25.33 kB

Running "uglify:issueList" (uglify) task
>> 1 file created 37.17 kB → 20.03 kB

Running "uglify:userActivity" (uglify) task
>> 1 file created 21.94 kB → 11.36 kB

Running "uglify:diagnose" (uglify) task
>> 1 file created 12.99 kB → 6.48 kB

Running "uglify:contributors" (uglify) task
>> 1 file created 1.05 kB → 737 B

Running "postcss:dist" (postcss) task
>> 1 processed stylesheet created.

Running "cssmin:combine" (cssmin) task
>> 1 file created. 269 kB → 59.45 kB

Done.

> webcompat@ start /Users/karl/code/webcompat.com
> source env/bin/activate || . env/bin/activate && python run.py

Traceback (most recent call last):
  File "run.py", line 41, in <module>
    raise ImportError('{0}\n\n{1}'.format(e, IMPORT_ERROR))
ImportError: cannot import name get_remote_address


==============================================
It seems like you don't have all dependencies.
Please re-run:
    pip install -r config/requirements.txt
==============================================


npm ERR! Darwin 17.2.0
npm ERR! argv "/Users/karl/.nvm/versions/node/v6.10.2/bin/node" "/Users/karl/.nvm/versions/node/v6.10.2/bin/npm" "run" "start"
npm ERR! node v6.10.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! webcompat@ start: `source env/bin/activate || . env/bin/activate && python run.py`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the webcompat@ start script 'source env/bin/activate || . env/bin/activate && python run.py'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the webcompat package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     source env/bin/activate || . env/bin/activate && python run.py
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs webcompat
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls webcompat
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/karl/code/webcompat.com/npm-debug.log
```

--- a/config/__init__.py
+++ b/config/__init__.py
@@ -17,9 +17,14 @@ import requests
 from environment import *  # nopep8
 from secrets import *  # nopep8
 
-MILESTONE_ERROR = '''It failed with {msg}!
+MILESTONE_ERROR = """It failed with {msg}!
 We will read from data/milestones.json.
-'''
+"""
+MILESTONE_MISSING_FILE = """Oooops.
+We can't find {path}
+Double check that everything is configured properly
+in config/secrets.py and try again. Good luck!
+"""
 
 
 def initialize_status():
@@ -49,8 +54,7 @@ def initialize_status():
         # Not working, let's use the cached copy
         # This might fail the first time.
         print(MILESTONE_ERROR.format(msg=error))
-        with open(milestones_path, 'r') as f:
-            milestones_content = f.read()
+        milestones_content = milestones_from_file(milestones_path)
     finally:
         # save in data/ the current version
         if milestones_content:
@@ -60,6 +64,17 @@ def initialize_status():
             return True
         else:
             return False
+
+
+def milestones_from_file(milestones_path):
+    """Attempt to read the milestones data from the filesystem."""
+    if os.path.isfile(milestones_path):
+        with open(milestones_path, 'r') as f:
+            milestones_content = f.read()
+        return milestones_content
+    else:
+        print(MILESTONE_MISSING_FILE.format(path=milestones_path))
+        return None
 
 
 def convert_milestones(milestones_content):

